python/kserve/test/test_inference_client.py

```diff
@@ -26,21 +28,33 @@
 
 @pytest.mark.asyncio
 class TestInferenceRESTClient:
-    @pytest.fixture(scope="class")
-    def rest_client(self, request):
+
+    @pytest_asyncio.fixture(scope="class")
+    async def app(self):
+        server = ModelServer()
+        rest_server = RESTServer(
+            kserve_app, server.dataplane, server.model_repository_extension
+        )
+        rest_server.create_application()
         model = DummyModel("TestModel")
         model.load()
         not_ready_model = DummyModel("NotReadyModel")
         # model.load()  # Model not loaded, i.e. not ready
-        server = ModelServer()
         server.register_model(model)
         server.register_model(not_ready_model)
-        rest_server = RESTServer(server.dataplane, server.model_repository_extension)
-        app = rest_server.create_application()
+        yield kserve_app
+        await server.model_repository_extension.unload("TestModel")
+        await server.model_repository_extension.unload("NotReadyModel")
+        kserve_app.routes.clear()
```

Comment:
These are executed after app is terminated?

Reply:
Yes, the statements after yield are used for cleanup